src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -710,7 +709,7 @@ public IEnumerable<HdAddress> CreateAddresses(Network network, int addressesQuan
         /// <param name="currentChainHeight">The current height of the chain. Used for calculating the number of confirmations a transaction has.</param>
         /// <param name="confirmations">The minimum number of confirmations required for transactions to be considered.</param>
         /// <returns>A collection of spendable outputs that belong to the given account.</returns>
-        public IEnumerable<UnspentOutputReference> GetSpendableTransactions(int currentChainHeight, int confirmations = 0)
+        public IEnumerable<UnspentOutputReference> GetSpendableTransactions(int currentChainHeight, NBitcoin.Consensus consensus, int confirmations = 0, bool includeImmature = false)
```

Comment:
Not sure if we need `includeImmature` parameter at all. Make make it always to be `false`? 

The method here is called `GetSpendableTransactions`, but the immature coins are never spendable. 

If we ever need to get those we might as well introduce a new method like `GetAllUnspentUTXOs` which will include immature coins.

Reply:
I agree, but correcting the behaviour of this method broke a lot of tests that were written under the old, incorrect behaviour. The `includeImmature` parameter is a temporary measure until these tests can be corrected (which I do not really see as being within the scope of the original bug fix).

Reply:
I think if the tests were testing the wrong logic and passing the tests were incorrect. 
We shouldn't adapt the code to allow incorrect behavior so the tests can pass.